lib/h3core.js

```diff
@@ -782,7 +792,7 @@ export function h3SetToMultiPolygon(h3Indexes, formatAsGeoJson) {
  * Compact a set of hexagons of the same resolution into a set of hexagons across
  * multiple levels that represents the same area.
  * @static
- * @param  {H3Index[]} h3Set H3 indexes to compact
+ * @param  {H3IndexInput[]} h3Set H3 indexes to compact
```

Comment:
Assuming that integer mode is primarily for perf, we might consider supporting a simple array of integers here (or even a `UInt32Array`), in `lower, upper, lower, upper, ...` order. This is just extending the polymorphism of the function, so we could add it later and it would still be backwards-compatible with the array-of-arrays input.

Reply:
Why not go "whole hog" with [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) and [BigUInt64Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigUint64Array)?

There is a [single example for Emscripten](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html?highlight=typedarray#access-memory-from-javascript) to access [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) data, so it should be relatively straightforward to efficiently pump that into the transpiled H3 and without the awkward syntax in Javascript (just put an `n` at the end of the integer and you're good).

The only downside is that apparently doesn't work in Safari yet? I wonder what their ETA is on fixing that, but Babel transpilation can be a temporary workaround.

Reply:
Sounds good - I think we'd continue with the polymorphic approach, so it shouldn't actually matter if some platforms don't support, since the type check would only succeed if the caller passed it in - so the burden is on the caller to determine platform compatibility issues, assuming we guard against the possibility that `BigInt` and `BigUInt64Array` may not be defined.

Again, not necessary for this PR, it would be easy to extend the same polymorphic approach to these input types.